### ISSUE TITLE
Question or Bug: Spiderable fails with `Error: Meteor code must always run within a Fiber.`

### ISSUE THREAD

**Issue by cramhead** [iron-meteor/iron-router#1069]
Seems spiderable fails in a dramatic sort of way. When pass an url that otherwise functions the `?_escaped_fragment_=` parameter. 
I'm wondering if others are having this issue?
I gather the iron-router is attempting to identify what to do with the parameter, but fails.

```
...
iron:dynamic-template                       1.0.0  
iron:layout                                 1.0.0  
iron:router                                 1.0.1  
... 
meteor-platform                             1.2.0  
...
spiderable                                  1.0.5  
...
```

```
 Error: Meteor code must always run within a Fiber. Try wrapping callbacks that you pass to non-Meteor libraries with Meteor.bindEnvironment.
   at Object.Meteor._nodeCodeMustBeInFiber (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/meteor.js:952:11)
   at [object Object]._.extend.get (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/meteor.js:964:12)
   at [object Object].RouteController.lookupOption (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/iron_router.js:139:29)
   at [object Object].Controller.extend.constructor (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/iron_router.js:99:21)
   at new Iron.utils.extend.ctor (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/iron_core.js:142:17)
   at Function.Route.createController (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/iron_router.js:504:18)
   at Function.Router.createController (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/iron_router.js:781:24)
   at Function.Router.dispatch (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/iron_router.js:1298:25)
   at Object.router (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/iron_router.js:611:12)
   at next (/Users/marc/.meteor/packages/webapp/.1.1.4.1x5frax++os+web.browser+web.cordova/npm/node_modules/connect/lib/proto.js:190:15)
   at Package (/Users/marc/Projects/METEOR/Sqwrl/app/.meteor/local/build/programs/server/packages/spiderable.js:163:11)
   at ChildProcess.exithandler (child_process.js:651:7)
   at ChildProcess.emit (events.js:98:17)
   at maybeClose (child_process.js:755:16)
   at Process._handle.onexit (child_process.js:822:5)
```

**Comment by timothyarmes** [comment]
I'm desperately trying to fix this problem myself. Here's what I've established:

1) The problem only occurs on the server.
2) It's not related to publications that aren't setting the ready state
3) It doesn't occur locally when I run using 'meteor run --production', so it isn't caused by incorrectly minified code
4) Both the server my local installation are running the same version of PhantomJS (1.9.8)

I'm at a loss. It's very hard to debug, and this is made worse since redeployment is slow. Any suggestions are welcome.

I haven't yet tried increasing the timeout since it's essentially instant on my local machine, I can't see why it would take more than 15 secs on the Digital Ocean server!

Are there any other known causes of this issue?

Tim

**Comment by maomorales** [comment]
Hey @timothyarmes - just check removing some packages that you have installed. There are packages that are not compatible and will cause this error to happen.

**Comment by timothyarmes** [comment]
Hi,

I will be trying that, but surely in that case it wouldn't work locally when run in production mode either?

Tim

--

**Comment by maomorales** [comment]
In my case it was working on local; and then in production didn't work. And it was one of the packages in conflict.

**Comment by eldog** [comment]
@timothyarmes I recommend giving the timeout a go because I had very similar problem: it was instant on my development machine then slow on a production linode (the only difference being that there is an Nginx proxy to my production app, maybe that's the slowdown) .

**Comment by timothyarmes** [comment]
I was about to go down the road of tracking down this issue, when I fell across this useful article:

http://www.meteorpedia.com/read/spiderable/#PhantomJS%20errors%20processing%20the%20page

I immediately s

**Comment by woniesong92** [comment]
Any update on this? I am getting the same error from `http://localhost:3000/?_escaped_fragment_=`

```
Error: Meteor code must always run within a Fiber. Try wrapping callbacks that you pass to non-Meteor libraries with Meteor.bindEnvironment.
    at Object.Meteor._nodeCodeMustBeInFiber (packages/meteor/dynamics_nodejs.js:9:1)
    at [object Object]._.extend.get (packages/meteor/dynamics_nodejs.js:21:1)
    at [object Object].RouteController.lookupOption (packages/iron:router/lib/route_controller.js:66:1)
    at new Controller.extend.constructor (packages/iron:router/lib/route_controller.js:26:1)
    at [object Object].ctor (packages/iron:core/lib/iron_core.js:88:1)
    at Function.Router.createController (packages/iron:router/lib/router.js:201:1)
    at Function.Router.dispatch (packages/iron:router/lib/router_server.js:39:1)
    at Object.router (packages/iron:router/lib/router.js:15:1)
    at next (/Users/user/.meteor/packages/webapp/.1.2.0.1kumrk0++os+web.browser+web.cordova/npm/node_modules/connect/lib/proto.js:190:15)
    at packages/spiderable/spiderable_server.js:123:1
```

**Comment by dutchakdev** [comment]
in my app DDP._allSubscriptionsReady() return false all time(

**Comment by mordka** [comment]
I had this problem, with a difference that it occurred on localhost as well. The problem was that PhantomJS in 1.X branch is not handling some JavaScript methods properly. I noticed it when  `phantomjs http://localhost:3000` resulted in parsing errors. 

On PhantomJS 2.0 spiderable runs seamlessly. The problem is that official 2.0 build is not released for Linux and it takes ages to compile it from sources. 
I grabbed 2.0 build from [here](https://github.com/Pyppe/phantomjs2.0-ubuntu14.04x64) and replaced it with `/usr/bin/phantomjs`. This eventually worked.
